### PR TITLE
Expand README, link to documentation reference page

### PR DESCRIPTION
**Pull request by pcholakov**
- add brief overview of included constructs and their purpose
- link to main CDK overview page in Restate docs (to be introduced in https://github.com/restatedev/documentation/pull/234)
- links to CDK example code elsewhere

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,4 +4,27 @@
 
 # Restate CDK support
 
-CDK construct library for deploying [Restate](https://restate.dev) and Restate services on AWS.
+AWS Cloud Development Kit (CDK) construct library for deploying [Restate](https://restate.dev) and Restate services on
+AWS. This library helps you when deploying Restate services to AWS Lambda as well as for managing a self-hosted Restate
```
Comment (tillrohrmann) [inline README.md:8]:
```suggestion
AWS. This library helps you when deploying Restate services to AWS Lambda as well as for managing self-hosted Restate
```

File: README.md [hunk 2]
```diff
@@ -4,4 +4,27 @@
...
+
+For a more detailed overview, please see the [Restate CDK documentation](https://docs.restate.dev/services/deployment/cdk).
+
+### Examples
+
+You can use the following examples as references for your own CDK projects:
+
+- [hello-world-lambda-cdk](https://github.com/restatedev/examples/tree/main/kotlin/hello-world-lambda-cdk) - provides a
+  simple example of a Lambda-deployed Kotlin handler
+- [Restate Holiday](https://github.com/restatedev/restate-holiday) - a more complex example of a fictional reservation
```
Comment (tillrohrmann) [inline README.md:29]: Off-topic: Are there any plans to unify the restate-holiday repo with the `examples` repo or do we want to keep it separate?

**Review by tillrohrmann — APPROVED** [review]
Thanks for creating this PR @pcholakov. LGTM. +1 for merging.

**Comment by pcholakov** [comment]
I'll merge only after the doc page CDK update has gone out.

**Comment by tillrohrmann** [comment]
Sounds good. Then let's also create a new release so that the README gets propagated to npmjs.

**Review by tillrohrmann — APPROVED** [review]

**Pull request merged**